synapse/state/v2.py

```diff
@@ -334,18 +341,28 @@ def _reverse_topological_power_sort(
     """
 
     graph = {}
-    for event_id in event_ids:
+    for idx, event_id in enumerate(event_ids):
         yield _add_event_and_auth_chain_to_graph(
             graph, room_id, event_id, event_map, state_res_store, auth_diff
         )
 
+        # We yield occasionally when we're working with large data sets to
+        # ensure that we don't block the reactor loop for too long.
+        if idx != 0 and idx % _YIELD_AFTER_ITERATIONS == 0:
```

Comment:
Idea: instead of checking `idx != 0`, why not make enumerate start at 1? The first yield would be 1 iteration early, but that's probably an acceptable tradeoff for avoiding a comparison on every iteration.

Reply:
Aha, didn't realise that was a thing `enumerate` supports